### PR TITLE
Fix "random" ECS task placement strategy

### LINKED ISSUE #5387: Random ordered_placement_strategy
_This issue was originally opened by @scherkan as hashicorp/terraform#18565. It was migrated here as a result of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

I can't attach random placement strategy to aws_ecs_service, according to:
https://www.terraform.io/docs/providers/aws/r/ecs_service.html#ordered_placement_strategy

### Terraform Version
Terraform v0.11.7
+ provider.aws v1.29.0
+ provider.template v1.0.0

```
...
```

### Terraform Configuration Files

```

resource "aws_ecs_service" "service" {
  name = "service"
  task_definition = "...."
  health_check_grace_period_seconds = 300
  cluster = "..."
  desired_count = 0
  launch_type = "EC2"

  load_balancer {
    target_group_arn = "..."
    container_name = "..."
    container_port = 8888
  }

  ordered_placement_strategy {
    type  = "spread"
    field = "attribute:ecs.availability-zone"
  }
  ordered_placement_strategy {
    type  = "spread"
    field = "instanceId"
  }

  ordered_placement_strategy {
    type = "random"
  }
  depends_on = ["aws_lb.internal"]
}
...
```

### Crash Output

aws_ecs_service.config-server-service: Creating...
  cluster:                                   "" => "arn:aws:ecs:eu-west-2:*********:cluster/cluster"
  deployment_maximum_percent:                "" => "200"
  deployment_minimum_healthy_percent:        "" => "100"
  desired_count:                             "" => "0"
  health_check_grace_period_seconds:         "" => "300"
  iam_role:                                  "" => "<computed>"
  launch_type:                               "" => "EC2"
  load_balancer.#:                           "" => "1"
  load_balancer.***.container_name:   "" => "docker"
  load_balancer.***.container_port:   "" => "8888"
  load_balancer.***.elb_name:         "" => ""
  load_balancer.***.target_group_arn: "" => "arn****...***"
  name:                                      "" => "service"
  ordered_placement_strategy.#:              "" => "3"
  ordered_placement_strategy.0.field:        "" => "attribute:ecs.availability-zone"
  ordered_placement_strategy.0.type:         "" => "spread"
  ordered_placement_strategy.1.field:        "" => "instanceId"
  ordered_placement_strategy.1.type:         "" => "spread"
  ordered_placement_strategy.2.type:         "" => "random"
  scheduling_strategy:                       "" => "REPLICA"
  task_definition:                           "" => "arn:aws:*******"

Error: Error applying plan:

1 error(s) occurred:

* aws_ecs_service.config-server-service: 1 error(s) occurred:

* aws_ecs_service.config-server-service: InvalidParameterException: random field should not be specified.
	status code: 400, request id: 5fb108e7........................ "service"

### Expected Behavior

Created  ECS service, with correct placement strategy:
- first AZ
- second instanceID
- random 
### Actual Behavior

Fail during apply script.
### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

### PR DESCRIPTION
**Pull request by jnjackins**
This change fixes the "random" ECS task placement strategy. Without this change, it is impossible to use this strategy. The AWS Go SDK requires the `Field` parameter to be `nil` when `Type` is random, but we were sending a pointer to `""`.

Fixes #5387

### CODE REVIEW

File: aws/resource_aws_ecs_service_test.go [hunk 1]
```diff
@@ -494,6 +494,15 @@ func TestAccAWSEcsService_withPlacementStrategy(t *testing.T) {
 					resource.TestCheckResourceAttr("aws_ecs_service.mongo", "ordered_placement_strategy.0.field", "memory"),
 				),
 			},
+			{
+				Config: testAccAWSEcsServiceWithRandomPlacementStrategy(clusterName, tdName, svcName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcsServiceExists("aws_ecs_service.mongo", &service),
+					resource.TestCheckResourceAttr("aws_ecs_service.mongo", "ordered_placement_strategy.#", "1"),
+					resource.TestCheckResourceAttr("aws_ecs_service.mongo", "ordered_placement_strategy.0.type", "random"),
+					resource.TestCheckNoResourceAttr("aws_ecs_service.mongo", "ordered_placement_strategy.0.field"),
```
Comment (bflad) [inline aws/resource_aws_ecs_service_test.go]: This seems to be failing acceptance testing at the moment:

```go
--- FAIL: TestAccAWSEcsService_withPlacementStrategy (32.95s)
    testing.go:538: Step 2 error: Check failed: Check 4/4 error: aws_ecs_service.mongo: Attribute 'ordered_placement_strategy.0.field' found when not expected
```

Comment (jnjackins) [reply aws/resource_aws_ecs_service_test.go]: Whoops, my mistake. Fixed and learned how to run the acceptance tests. :)

We want Field attribute to be nil when creating the resource, but the attribute still exists when querying the resource.

**Comment by bflad** [comment]
This has been released in [version 1.41.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1410-october-18-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Comment by jnjackins** [comment]
Excellent, thanks!

**Pull request merged**